public/app/plugins/datasource/logging/result_transformer.ts

```diff
@@ -32,21 +88,89 @@ export function processEntry(entry: { line: string; timestamp: string }, stream)
     key,
     logLevel,
     timeFromNow,
+    timeJs,
     timeLocal,
     entry: line,
+    labels: formatLabels(labels),
     searchWords: [stream.search],
     timestamp: timestamp,
   };
 }
 
-export function processStreams(streams, limit?: number): LogsModel {
+export function mergeStreams(streams: LogsStream[], limit?: number): LogsModel {
+  // Find meta data
+  const commonLabels = findCommonLabels(streams.map(stream => stream.parsedLabels));
+  const meta: LogsMetaItem[] = [
+    {
+      label: 'Common labels',
+      value: formatLabels(commonLabels),
+    },
+  ];
+
+  // Flatten entries of streams
   const combinedEntries = streams.reduce((acc, stream) => {
-    return [...acc, ...stream.entries.map(entry => processEntry(entry, stream))];
+    // Overwrite labels to be only the non-common ones
+    const labels = formatLabels(findUncommonLabels(stream.parsedLabels, commonLabels));
+    return [
+      ...acc,
+      ...stream.entries.map(entry => ({
+        ...entry,
+        labels,
+      })),
+    ];
   }, []);
+
+  const commonLabelsAlias =
+    streams.length === 1 ? formatLabels(commonLabels) : `Stream with common labels ${formatLabels(commonLabels)}`;
+  const series = streams.map((stream, index) => {
+    const colorIndex = index % colors.length;
+    stream.graphSeries.setColor(colors[colorIndex]);
+    stream.graphSeries.alias = formatLabels(findUncommonLabels(stream.parsedLabels, commonLabels), commonLabelsAlias);
+    return stream.graphSeries;
+  });
+
   const sortedEntries = _.chain(combinedEntries)
     .sortBy('timestamp')
     .reverse()
     .slice(0, limit || combinedEntries.length)
     .value();
-  return { rows: sortedEntries };
+
+  meta.push({
+    label: 'Limit',
+    value: `${limit} (${sortedEntries.length} returned)`,
+  });
+
+  return { meta, series, rows: sortedEntries };
+}
+
+export function processStream(stream: LogsStream, limit?: number): LogsStream {
```

Comment:
unit tests for this function would be good, unless this is still POC phase stuff you want to rewrite. but then we might be need to create issues for that so we remember it.

Reply:
Will add those later.